dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/internal/persistence/file/datastore/DataStoreFileHandlerTest.kt

```diff
@@ -0,0 +1,179 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.file.datastore
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.storage.datastore.DataStoreCallback
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.datadog.android.core.internal.persistence.datastore.DataStoreFileHandler
+import com.datadog.android.core.internal.persistence.datastore.DatastoreFileReader
+import com.datadog.android.core.internal.persistence.datastore.DatastoreFileWriter
+import com.datadog.android.core.persistence.Serializer
+import com.datadog.android.core.persistence.datastore.DataStoreContent
+import com.datadog.android.utils.forge.Configurator
+import fr.xgouchet.elmyr.annotation.IntForgery
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.doAnswer
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Future
+import java.util.concurrent.TimeUnit
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DataStoreFileHandlerTest {
+
+    private lateinit var testedDataStoreHandler: DataStoreFileHandler
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Mock
+    lateinit var mockExecutorService: ExecutorService
+
+    @Mock
+    lateinit var mockDataStoreFileReader: DatastoreFileReader
+
+    @Mock
+    lateinit var mockDeserializer: Deserializer<String, ByteArray>
+
+    @Mock
+    lateinit var mockDatastoreFileWriter: DatastoreFileWriter
+
+    @Mock
+    lateinit var mockSerializer: Serializer<String>
+
+    @StringForgery
+    lateinit var fakeFeatureName: String
+
+    @StringForgery
+    lateinit var fakeKey: String
+
+    @StringForgery
+    lateinit var fakeDataString: String
+
+    private lateinit var fileCallback: DataStoreCallback<ByteArray>
+
+    @BeforeEach
+    fun setup() {
+        whenever(mockExecutorService.submit(any())) doAnswer {
+            it.getArgument<Runnable>(0).run()
+            StubFuture()
+        }
```

Comment:
If you use `executeSafe` in the `DataStoreFileHandler`, `StubFuture` is not needed.

```suggestion
        whenever(mockExecutorService.execute(any())) doAnswer {
            it.getArgument<Runnable>(0).run()
        }
```